addons/account/models/account_partial_reconcile.py

```diff
@@ -135,9 +136,21 @@ def unlink(self):
     @api.model_create_multi
     def create(self, vals_list):
         partials = super().create(vals_list)
+        self._pay_matched_payment(partials)
         self._update_matching_number(partials.debit_move_id + partials.credit_move_id)
         return partials
 
+    @api.model
+    def _pay_matched_payment(self, partials):
+        for partial in partials:
+            amount = partial.credit_amount_currency
+            matched_payments = partial.credit_move_id.move_id.matched_payment_ids | partial.debit_move_id.move_id.matched_payment_ids
+            decimal_places = partial.credit_currency_id.decimal_places
+            for matched_payment in matched_payments:
+                if not float_compare(matched_payment.amount, amount, precision_digits=decimal_places) and matched_payment.state != 'paid':
```

Comment:
Be careful that the amount field doesn't take the direction of the transaction (in/outbound) into account

Even if it is not likely to register a negative payment it would be more correct to take care of the sign

Reply:
What do you mean by "take care of the sign" ?
If the `matched_payment.amount` is negative, should we match it with the partial? I mean currently it won't match which I think makes sense, no?